hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/TestAutoKeyGenForSQL.scala

```diff
@@ -97,4 +99,198 @@ class TestAutoKeyGenForSQL extends SparkClientFunctionalTestHarness {
+         | ts BIGINT,
+         | uuid STRING,
+         | rider STRING,
+         | driver STRING,
+         | fare DOUBLE,
+         | city STRING )
+         | USING hudi
+         | OPTIONS (
+         |  hoodie.metadata.enable = 'true',
+         |  hoodie.enable.data.skipping = 'true',
+         |  hoodie.write.record.merge.mode = 'COMMIT_TIME_ORDERING',
+         |  hoodie.clean.commits.retained = '5',
+         |  hoodie.keep.max.commits = '3',
+         |  hoodie.keep.min.commits = '2',
+         |  hoodie.clustering.inline = 'true',
+         |  hoodie.clustering.inline.max.commits = '2',
+         |  hoodie.compact.inline = '$compactionEnabled')
+         | PARTITIONED BY(city)
+         | LOCATION '$basePath'
+         | TBLPROPERTIES (hoodie.datasource.write.table.type='${tableType.name}')
+       """.stripMargin)
+    spark.sql(
+      s"""
+         |INSERT INTO $tableName VALUES
+         |  (1695159649087,'334e26e9-8355-45cc-97c6-c31daf0df330','rider-A','driver-K',19.10,'san_francisco'),
+         |  (1695091554788,'e96c4396-3fad-413a-a942-4cb36106d721','rider-B','driver-M',27.70 ,'san_francisco'),
+         |  (1695046462179,'9909a8b1-2d15-4d3d-8ec9-efc48c536a00','rider-C','driver-L',33.90 ,'san_francisco'),
+         |  (1695332066204,'1dced545-862b-4ceb-8b43-d2a568f6616b','rider-D','driver-O',93.50,'san_francisco'),
+         |  (1695516137016,'e3cf430c-889d-4015-bc98-59bdce1e530c','rider-E','driver-P',34.15,'sao_paulo'),
+         |  (1695376420876,'7a84095f-737f-40bc-b62f-6b69664712d2','rider-F','driver-Q',43.40 ,'sao_paulo'),
+         |  (1695173887231,'3eeb61f7-c2b0-4636-99bd-5d7a5a1d2c04','rider-G','driver-S',41.06 ,'chennai'),
+         |  (1695115999911,'c8abbe79-8d89-47ea-b4ce-4d224bae5bfa','rider-H','driver-T',17.85,'chennai');
+      """.stripMargin)
+
+    import java.util.UUID
+    for (i <- 0 until 10) {
+      val ts: Long = 1695115999911L + i + 1
+      val uuid: String = UUID.randomUUID.toString
+      val rider: String = s"rider-$i"
+      val driver: String = s"driver-$i"
+      val fare: Float = Random.nextFloat
+      val city: String = randomString(8)
+      spark.sql(
+        s"""
+           |INSERT INTO $tableName VALUES
+           |($ts, '$uuid', '$rider', '$driver', $fare, '$city');
+        """.stripMargin)
+    }
+
+    for (i <- 0 until 10) {
+      val ts: Long = 1695115999911L + i + 1
+      val rider: String = s"rider-${'A' + new Random().nextInt(8)}"
+      spark.sql(
+        s"""
+           |UPDATE $tableName
+           |SET ts = $ts
+           |WHERE rider = '$rider'
+        """.stripMargin)
+    }
+
+    for (i <- 0 until 2) {
+      val rider: String = s"rider-${('A' + new Random().nextInt(8)).toChar}"
+      spark.sql(
+        s"""
+           |DELETE FROM $tableName
+           |WHERE rider = '$rider'
+        """.stripMargin)
+    }
+
+    // Validate: data integrity
+    val noRecords = spark.sql(s"SELECT * FROM $tableName").count()
+    assertEquals(16, noRecords)
```

Comment:
can we declare a set for list of riders to update and delete. just to ensure the random does not produce the same rider records again. 
so we know for sure, we are doing to update or delete two diff rider entries.